packages/canvas-panel-cookbook/src/pages/SlideShow/components/SwappableView.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import * as PropTypes from 'prop-types';
+
+import {
+  StaticImageViewport,
+  FullPageViewport,
+  OpenSeadragonViewport,
+  SingleTileSource,
+  withBemClass,
+} from '@canvas-panel/core';
+
+import './SwappableView.scss';
+
+const StaticSlideViewportBounds = props => (
+  <div className="slide-cover">{props.children}</div>
+);
+
+class SwappableView extends Component {
+  state = {
+    isInteractive: false,
+    itemWidth: 0,
+    itemHeight: 0,
+  };
+
+  constructor(props) {
+    super(props);
+    this.setViewportToStatic = this.setViewportToStatic.bind(this);
+    this.setViewportToInteractive = this.setViewportToInteractive.bind(this);
+  }
+  static propTypes = {
+    exitInteractiveModeButtonLabel: PropTypes.string,
+    enterInteractiveModeButtonLabel: PropTypes.string,
+    canvas: PropTypes.any,
+    manifest: PropTypes.any,
+  };
+
+  static defaultProps = {
+    exitInteractiveModeButtonLabel: 'Exit Interactive Mode',
+    enterInteractiveModeButtonLabel: 'Explore',
+  };
+
+  setViewportToStatic() {
+    this.setState({
+      isInteractive: false,
+    });
+  }
+
+  setViewportToInteractive() {
```

Comment:
you can use class props to avoid binding in the constructor:
```js
setViewportToInteractive = () => ...
```

To stay consistent with the other components in the project.